lib/transaction.js

```diff
@@ -45,6 +46,12 @@ class Transaction extends EventEmitter {
       outerTx ? 'nested' : 'top level'
     );
 
+    // `this` can potentially serve as an `outerTx` for another
+    // Transaction.  So, go ahead and establish `_lastChild` now.
+    this._lastChild = Promise.resolve();
```

Comment:
When do we overwrite this?

Reply:
It gets overwritten towards the bottom of each child Transaction's constructor:

```
outerTx._lastChild = basePromise.catch(() => {});
```

BTW: I think I'm gonna mark this PR as `WIP:` for now.  I'm still thinking about extracting the `outerTx` thing a bit further.